packages/core/src/dialects/abstract/query-interface.js

```diff
@@ -155,7 +155,7 @@ export class AbstractQueryInterface extends AbstractQueryInterfaceTypeScript {
   async createTable(tableName, attributes, options, model) {
     options = { ...options };
 
-    if (model) {
+    if (model && this.queryGenerator.dialect.supports.uniqueKeys) {
```

Comment:
The `uniqueKeys` getter in model does `Object.create(null)` so this option is always filled.

This check might not be the nicest solution, but it works fine